src/lib/idl.ml

```diff
@@ -104,6 +104,20 @@ let get_wire_name description name =
     | None -> name)
 
 
+let check_dups_exn api params =
+  let named = params |> List.filter_map Fun.id |> List.rev in
```

Comment:
The list gets sorted later and otherwise only the length is taken - so we need the `List.rev` only for the error reporting (where we could also apply it).

Could we move the `params_of` call from the call sites into this function? Currently this function is neither a universal list function, nor very domain specific.